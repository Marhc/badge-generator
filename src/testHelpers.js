// @flow

const toString = value => Object.prototype.toString.apply(value);

export function findByType(testInstance: ReactTestInstance, Type: Object) {
  const { instance } = testInstance.findByType(Type);
  if (!(instance instanceof Type)) {
    throw new TypeError(
      `Expected app to be an instance of ${Type.name}, got ${toString(
        instance,
      )}`,
    );
  }
  return instance;
}
