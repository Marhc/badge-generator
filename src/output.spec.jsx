// @flow

import React from 'react';
import TestRenderer from 'react-test-renderer';
import OutputWithStyle, { Output } from './output';
import { findByType } from './testHelpers';

describe('Output', () => {
  describe('when services are empty', () => {
    it('renders', () => {
      const component = TestRenderer.create(
        <OutputWithStyle
          services={[]}
          badgeStyle="STYLE"
          repository="REPOSITORY"
        />,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('when services are set', () => {
    it('renders', () => {
      const services = [
        {
          name: 'GitHub stars',
          url: 'https://github.com/{repository}',
          imageUrl: 'https://img.shields.io/github/stars/{repository}.svg',
          title: 'Build Status',
        },
        {
          name: 'Travis CI',
          url: 'https://travis-ci.org/{repository}',
          imageUrl: 'https://img.shields.io/travis/{repository}/{branch}.svg',
          title: 'Build Status',
        },
      ];
      const component = TestRenderer.create(
        <OutputWithStyle
          services={services}
          badgeStyle="STYLE"
          repository="REPOSITORY"
        />,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('#handleFormatChange()', () => {
    const classes = {
      subtitle: '',
      badges: '',
      badge: '',
      formats: '',
      radioGroup: '',
      code: '',
    };

    it('is a function', () => {
      const renderer = TestRenderer.create(
        <Output
          services={[]}
          badgeStyle="STYLE"
          repository="REPOSITORY"
          classes={classes}
        />,
      );
      const component = findByType(renderer.root, Output);
      expect(component.handleFormatChange).toBeFunction();
    });

    it("updates Output 'formatIndex' state", () => {
      const renderer = TestRenderer.create(
        <Output
          services={[]}
          badgeStyle="STYLE"
          repository="REPOSITORY"
          classes={classes}
        />,
      );
      const component = findByType(renderer.root, Output);
      expect(component.state.formatIndex).not.toBe(1);

      component.handleFormatChange({}, '1');
      expect(component.state.formatIndex).toBe(1);
    });
  });
});
