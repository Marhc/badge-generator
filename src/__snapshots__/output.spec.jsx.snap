// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output when services are empty renders its components 1`] = `
<React.Fragment>
  <WithStyles(Typography)
    className=""
    headlineMapping={
      Object {
        "h6": "h2",
      }
    }
    variant="h6"
  >
    Preview
  </WithStyles(Typography)>
  <p
    className=""
  />
  <WithStyles(Typography)
    className=""
    headlineMapping={
      Object {
        "h6": "h2",
      }
    }
    variant="h6"
  >
    Source code
  </WithStyles(Typography)>
  <form
    autoComplete="off"
    className=""
    noValidate={true}
  >
    <RadioGroup
      aria-label="Format"
      className=""
      onChange={[Function]}
      value="2"
    >
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="URL"
        title="URL"
        value="0"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Image URL"
        title="Image URL"
        value="1"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Markdown"
        title="Markdown"
        value="2"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Org Mode"
        title="Org Mode"
        value="3"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Textile"
        title="Textile"
        value="4"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Rdoc"
        title="Rdoc"
        value="5"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="AsciiDoc"
        title="AsciiDoc"
        value="6"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="RST"
        title="RST"
        value="7"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Pod"
        title="Pod"
        value="8"
      />
    </RadioGroup>
  </form>
  <pre
    className=""
  />
</React.Fragment>
`;

exports[`Output when services are set renders its components 1`] = `
<React.Fragment>
  <WithStyles(Typography)
    className=""
    headlineMapping={
      Object {
        "h6": "h2",
      }
    }
    variant="h6"
  >
    Preview
  </WithStyles(Typography)>
  <p
    className=""
  >
    <a
      className=""
      href="https://github.com/REPOSITORY"
      title="Build Status"
    >
      <img
        alt="Build Status"
        src="https://img.shields.io/github/stars/REPOSITORY.svg?style=STYLE"
      />
    </a>
    <a
      className=""
      href="https://travis-ci.org/REPOSITORY"
      title="Build Status"
    >
      <img
        alt="Build Status"
        src="https://img.shields.io/travis/REPOSITORY/master.svg?style=STYLE"
      />
    </a>
  </p>
  <WithStyles(Typography)
    className=""
    headlineMapping={
      Object {
        "h6": "h2",
      }
    }
    variant="h6"
  >
    Source code
  </WithStyles(Typography)>
  <form
    autoComplete="off"
    className=""
    noValidate={true}
  >
    <RadioGroup
      aria-label="Format"
      className=""
      onChange={[Function]}
      value="2"
    >
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="URL"
        title="URL"
        value="0"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Image URL"
        title="Image URL"
        value="1"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Markdown"
        title="Markdown"
        value="2"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Org Mode"
        title="Org Mode"
        value="3"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Textile"
        title="Textile"
        value="4"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Rdoc"
        title="Rdoc"
        value="5"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="AsciiDoc"
        title="AsciiDoc"
        value="6"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="RST"
        title="RST"
        value="7"
      />
      <WithStyles(WithFormControlContext(FormControlLabel))
        control={<WithStyles(Radio) />}
        label="Pod"
        title="Pod"
        value="8"
      />
    </RadioGroup>
  </form>
  <pre
    className=""
  >
    [![Build Status](https://img.shields.io/github/stars/REPOSITORY.svg?style=STYLE)](https://github.com/REPOSITORY)

    [![Build Status](https://img.shields.io/travis/REPOSITORY/master.svg?style=STYLE)](https://travis-ci.org/REPOSITORY)

  </pre>
</React.Fragment>
`;
